src/server/pfs/server/driver_new_storage.go

```diff
@@ -45,45 +44,53 @@ func (d *driver) finishCommitNewStorageLayer(txnCtx *txnenv.TransactionContext,
 	if description != "" {
 		commitInfo.Description = description
 	}
-	// Compact the commit changes into a diff file set.
-	commitPath := path.Join(commit.Repo.Name, commit.ID)
-	if err := d.compact(txnCtx.Client.Ctx(), path.Join(commitPath, fileset.Diff), []string{commitPath}); err != nil {
-		return err
-	}
-	// Compact the commit changes (diff file set) into the total changes in the commit's ancestry.
-	var compactSpec *fileset.CompactSpec
-	if commitInfo.ParentCommit == nil {
-		compactSpec, err = d.storage.CompactSpec(txnCtx.Client.Ctx(), commitPath)
-	} else {
-		// (bryce) how to handle parent commit that is not closed?
-		parentCommitPath := path.Join(commitInfo.ParentCommit.Repo.Name, commitInfo.ParentCommit.ID)
-		compactSpec, err = d.storage.CompactSpec(txnCtx.Client.Ctx(), commitPath, parentCommitPath)
-	}
-	if err != nil {
-		return err
-	}
-	if err := d.compact(txnCtx.Client.Ctx(), compactSpec.Output, compactSpec.Input); err != nil {
-		return err
-	}
-	// (bryce) need size.
-	commitInfo.SizeBytes = uint64(0)
-	commitInfo.Finished = now()
-	return d.writeFinishedCommit(txnCtx.Stm, commit, commitInfo)
+	// Run compaction task.
+	return d.compactionQueue.RunTaskBlock(txnCtx.Client.Ctx(), func(m *work.Master) error {
+		commitPath := path.Join(commit.Repo.Name, commit.ID)
+		// (bryce) need some cleanup improvements, probably garbage collection.
+		// (bryce) this should be a retry when garbage collection is integrated.
+		d.storage.Delete(context.Background(), path.Join(commitPath, fileset.Diff))
+		d.storage.Delete(context.Background(), path.Join(commitPath, fileset.Compacted))
```

Comment:
Why are these using `context.Background()` when the above uses `txnCtx.Client.Ctx()`?

Reply:
This is a hack to get compactions working before garbage collection is integrated. Basically, we need to clean up the diff and compacted file set for failed compactions. This will be handled properly when garbage collection is integrated.